
As reported in:
https://bugs.mageia.org/show_bug.cgi?id=27936

adding Mageia version branding, breaks loading the VBoxDtrace extension with the error:

Failed to load
'/usr/lib64/virtualbox/ExtensionPacks/Oracle_VBoxDTrace_Extension_Pack.xml':
Invalid version string: 6.1.16_Mageia

Fix it up by using the _NO_PUB variant when generating the xml.

Signed-off-by: Thomas Backlund <tmb@mageia.org>

---
 Config.kmk                               |    1 +
 src/VBox/ExtPacks/VBoxDTrace/ExtPack.xml |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff -Nurp VirtualBox-7.0.2.orig/Config.kmk VirtualBox-7.0.2/Config.kmk
--- VirtualBox-7.0.2.orig/Config.kmk	2022-10-19 21:17:13.000000000 +0300
+++ VirtualBox-7.0.2/Config.kmk	2022-10-22 15:12:37.178853337 +0300
@@ -3798,6 +3798,7 @@ VBOX_EDIT_VERSION_CMD_FN = $(SED) \
 	-e 's+@VBOX_VERSION_MINOR@+$(VBOX_VERSION_MINOR)+g' \
 	-e 's+@VBOX_VERSION_BUILD@+$(VBOX_VERSION_BUILD)+g' \
 	-e 's+@VBOX_VERSION_STRING@+$(VBOX_VERSION_STRING)+g' \
+	-e 's/@VBOX_VERSION_STRING_NO_PUB@/$(VBOX_VERSION_STRING_NO_PUB)/g' \
 	-e 's+@VBOX_SVN_REV@+$(VBOX_SVN_REV)+g' \
 	-e 's+@VBOX_VENDOR@+$(VBOX_VENDOR)+g' \
 	-e 's+@VBOX_VENDOR_SHORT@+$(VBOX_VENDOR_SHORT)+g' \
diff -Nurp VirtualBox-7.0.2.orig/src/VBox/ExtPacks/VBoxDTrace/ExtPack.xml VirtualBox-7.0.2/src/VBox/ExtPacks/VBoxDTrace/ExtPack.xml
--- VirtualBox-7.0.2.orig/src/VBox/ExtPacks/VBoxDTrace/ExtPack.xml	2022-10-19 21:27:27.000000000 +0300
+++ VirtualBox-7.0.2/src/VBox/ExtPacks/VBoxDTrace/ExtPack.xml	2022-10-22 15:10:10.042857721 +0300
@@ -2,7 +2,7 @@
 <VirtualBoxExtensionPack xmlns="http://www.virtualbox.org/VirtualBoxExtensionPack" version="1.0">
     <Name>Oracle VBoxDTrace Extension Pack</Name>
     <Description>Experimental and unsupported extension pack providing DTrace features to VirtualBox.</Description>
-    <Version revision="@VBOX_SVN_REV@">@VBOX_VERSION_STRING@</Version>
+    <Version revision="@VBOX_SVN_REV@">@VBOX_VERSION_STRING_NO_PUB@</Version>
     <MainModule>VBoxDTraceMain</MainModule>
 </VirtualBoxExtensionPack>
 
